c10/core/impl/alloc_cpu.cpp

```diff
@@ -81,6 +113,17 @@ void* alloc_cpu(size_t nbytes) {
       " (",
       strerror(err),
       ")");
+#ifdef __linux__
+  // MADV_HUGEPAGE advise is available only for linux.
+  // general posix compliant systems can check POSIX_MADV_SEQUENTIAL advise.
+  if (is_thp_alloc(nbytes)) {
+    int ret = madvise(data, nbytes, MADV_HUGEPAGE);
+    if (ret != 0) {
+      TORCH_WARN_ONCE(
+          "thp madvise for HUGEPAGE failed with %s\n", strerror(errno));
```

Comment:
TORCH_WARN_ONCE does not take format strings, but rather concatenates output
```suggestion
          "thp madvise for HUGEPAGE failed with ", strerror(errno));
```